wordpress-mobile/WordPress-Android
# Gutenberg - Audio Block Processor

## Findings
This PR is the Android side of the [iOS implementation for updating the HTML content](https://github.com/wordpress-mobile/WordPress-iOS/pull/15420) of the Audio Block without Gutenberg being open. 

## Solution
The `AudioBlockProcessor` replaces the `<audio src>` tag that contains local file URL with a remote media URL once the file upload is complete. The id of the Audio Block is also updated with a remote variant as well. 

## Testing

1. Add an Audio block to a new post.
2. Upload an audio file to the block (I  lessened the Cellular Network type and Signal strength to replicate a slow network so I  had enough time to close the editor).
3. Close the editor by pressing the back button.
4. Let the upload to finish in the background.
5. When the audio file is uploaded and the post saved, open the post again (on web or mobile).

Verify that the behavior is functioning correctly by :
  - Checking that the block renders properly.
  - Checking that all references to src are updated to the remote URL.
  - Checking that the media ID has been updated to the remote one.

## Reviewing 
* Only 1 reviewer is needed but anyone can review. 
* This PR depends on the [Consolidated Media Picker Audio Block Integration PR](https://github.com/wordpress-mobile/WordPress-Android/pull/13612) so that has to be merged before this is merged in.

## Submitter Checklist
